### PR TITLE
Release notes

### LINKED ISSUE #375: Create a mechanism for adding notes to the changelog
**Is your feature request related to a problem? Please describe.**

Currently the changelog is generated by PR titles. I think for the vast majority of use cases this is sufficient. In some cases though, there may be a need to add context to the release.

An example would be for breaking changes. In those circumstances, having some sort of migration guide as a part of the changelog/release would be beneficial. 

**Describe the solution you'd like**

I'm not 100% on the setup here, but it could be as simple as having a release notes section in your PR description

```
# Release Notes

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
```

or maybe a `details` tag

```
<details>
  <summary>Release Notes</summary

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

<details>
```

In the output it could have a top level heading like `### Release Notes` and it might be useful to separate each set of docs by PR. 

The final rendered output of the changelog might look something like the below: 

---

# v5.0.1 (Sat May 04 2019)

### Release Notes

_From #371_

`auto shipit` will can now only be ran on the base branch ([which is configurable]()). If you want to ship a version in the PR, you can now look to using [canary releases]().

---

#### 🐛  Bug Fix

- shipit will publish a canary locally when not on master [#371](https://github.com/intuit/auto/pull/371) ([@hipstersmoothie](https://github.com/hipstersmoothie))

#### Authors: 1

- Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie))

### PR DESCRIPTION
**Pull request by hipstersmoothie**
# What Changed

add a special markdown section in PR that get added as release notes to the changelog

# Why

closes #375 

Todo:

- [x] Add tests
- [x] Add docs
- [ ] Add yourself to contributors (run `yarn contributors:add`)

Published PR with canary version: `6.3.0-canary.380.4902`

### CODE REVIEW

**Comment by adierkens** [comment]
Published PR with canary version: `6.1.0-canary.380.4705`

File: src/changelog.ts [hunk 1]
```diff
@@ -282,4 +276,63 @@ export default class Changelog {
...
+    let section = '';
+
+    commits.map(commit => {
+      const pr = commit.pullRequest;
+
+      if (!pr || !pr.body) {
+        return;
+      }
+
+      const title = /^> [#]{0,5} Release Notes$/;
```
Comment (zephraph) [inline src/changelog.ts]: This match is pretty strict. I think it'd be worth loosening the whitespace requirements and making it case insensitive.

File: src/__tests__/__snapshots__/changelog.test.ts.snap [hunk 2]
```diff
@@ -19,6 +19,30 @@ exports[`Hooks title 1`] = `
...
+exports[`generateReleaseNotes should add additional release notes 1`] = `
+"### Release Notes
+
+_From #1235_
+
+Here is how you upgrade
+
+---"
+`;
+
```
Comment (zephraph) [inline src/__tests__/__snapshots__/changelog.test.ts.snap]: 😍

**Comment by adierkens** [comment]
:rocket: PR was released in v6.3.0 :rocket:

**Pull request merged**